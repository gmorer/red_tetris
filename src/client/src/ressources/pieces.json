[
	{
		"name": "Bâton",
		"color": "#FF765C",
		"position": [
			{
				"display": [
					["x", "x", "x", "x"],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 0,
					"top": 0,
					"bot": 3
				}
			},
			{
				"display": [
					[" ", "x", " ", " "],
					[" ", "x", " ", " "],
					[" ", "x", " ", " "],
					[" ", "x", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 2,
					"top": 0,
					"bot": 0
				}
			}
		]
	},
	{
		"name": "Bloc",
		"color": "#E85654",
		"position": [
			{
				"display": [
					[" ", "x", "x", " "],
					[" ", "x", "x", " "],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 1,
					"top": 0,
					"bot": 2
				}
			}
		]
	},
	{
		"name": "Té",
		"color": "#FF68AC",
		"position": [
			{
				"display": [
					[" ", "x", " ", " "],
					[" ", "x", "x", " "],
					[" ", "x", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 1,
					"top": 0,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", " ", " ", " "],
					["x", "x", "x", " "],
					[" ", "x", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 1,
					"top": 1,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", "x", " ", " "],
					["x", "x", " ", " "],
					[" ", "x", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 2,
					"top": 0,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", "x", " ", " "],
					["x", "x", "x", " "],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 1,
					"top": 0,
					"bot": 2
				}
			}
		]
	},
	{
		"name": "Lambda",
		"color": "#E854E1",
		"position": [
			{
				"display": [
					[" ", " ", " ", " "],
					["x", "x", "x", " "],
					["x", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 1,
					"top": 1,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", "x", "x", " "],
					[" ", " ", "x", " "],
					[" ", " ", "x", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 1,
					"top": 0,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", " ", " ", " "],
					[" ", " ", " ", "x"],
					[" ", "x", "x", "x"],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 0,
					"top": 1,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", "x", " ", " "],
					[" ", "x", " ", " "],
					[" ", "x", "x", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 1,
					"top": 0,
					"bot": 1
				}
			}
		]
	},
	{
		"name": "Gamma",
		"color": "#D75CFF",
		"position": [
			{
				"display": [
					["x", "x", "x", " "],
					[" ", " ", "x", " "],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 1,
					"top": 0,
					"bot": 2
				}
			},
			{
				"display": [
					[" ", " ", "x", " "],
					[" ", " ", "x", " "],
					[" ", "x", "x", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 1,
					"top": 0,
					"bot": 1
				}
			},
			{
				"display": [
					[" ", " ", " ", " "],
					["x", " ", " ", " "],
					["x", "x", "x", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 1,
					"top": 1,
					"bot": 1
				}
			},
			{
				"display": [
					["x ", "x", " ", " "],
					["x", " ", " ", " "],
					["x", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 2,
					"top": 0,
					"bot": 1
				}
			}
		]
	},
	{
		"name": "Biais",
		"color": "#E38986",
		"position": [
			{
				"display": [
					["x", "x", " ", " "],
					[" ", "x", "x", " "],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 1,
					"top": 0,
					"bot": 2
				}
			},
			{
				"display": [
					[" ", "x", " ", " "],
					["x", "x", " ", " "],
					["x", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 2,
					"top": 0,
					"bot": 1
				}
			}
		]
	},
	{
		"name": "Biais inversé",
		"color": "#C888FF",
		"position": [
			{
				"display": [
					[" ", "x", "x", " "],
					["x", "x", " ", " "],
					[" ", " ", " ", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 0,
					"right": 2,
					"top": 0,
					"bot": 2
				}
			},
			{
				"display": [
					[" ", "x", " ", " "],
					[" ", "x", "x", " "],
					[" ", " ", "x", " "],
					[" ", " ", " ", " "]
				],
				"hitbox": {
					"left": 1,
					"right": 2,
					"top": 0,
					"bot": 1
				}
			}
		]
	}
]