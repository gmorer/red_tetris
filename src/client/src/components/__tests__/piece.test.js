import Piece from './../piece';
import React from 'react';
import renderer from 'react-test-renderer';
import pieces from '../../ressources/pieces'


test('Some test', () => {
	const component = renderer.create(<Piece pos={{ x: 6, y: 7, rotation: 1 }} piece={pieces[0]} isPreview={false} />)
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})