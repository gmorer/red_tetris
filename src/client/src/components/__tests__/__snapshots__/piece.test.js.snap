// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some test 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "20%",
      "left": "60%",
      "position": "relative",
      "top": "35%",
      "width": "40%",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "25%",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "border": "2px solid #5f1522",
          "boxSizing": "border-box",
          "height": "100%",
          "width": "25%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "borderColor": "#9f233a red red #be2a45",
              "borderStyle": "solid",
              "borderWidth": "10em 0em 0em 10em",
              "height": "0px",
              "width": "0px",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "25%",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "border": "2px solid #5f1522",
          "boxSizing": "border-box",
          "height": "100%",
          "width": "25%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "borderColor": "#9f233a red red #be2a45",
              "borderStyle": "solid",
              "borderWidth": "10em 0em 0em 10em",
              "height": "0px",
              "width": "0px",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "25%",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "border": "2px solid #5f1522",
          "boxSizing": "border-box",
          "height": "100%",
          "width": "25%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "borderColor": "#9f233a red red #be2a45",
              "borderStyle": "solid",
              "borderWidth": "10em 0em 0em 10em",
              "height": "0px",
              "width": "0px",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "25%",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "border": "2px solid #5f1522",
          "boxSizing": "border-box",
          "height": "100%",
          "width": "25%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "borderColor": "#9f233a red red #be2a45",
              "borderStyle": "solid",
              "borderWidth": "10em 0em 0em 10em",
              "height": "0px",
              "width": "0px",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "height": "100%",
          "width": "25%",
        }
      }
    />
  </div>
</div>
`;
